haskellPackages.apecs-gloss fails to build

**Describe the bug**
`haskellPackagea.apecs-gloss` fails to build with error

```
running nix build -f. haskellPackages.apecs-gloss
builder for '/nix/store/vwr9wdbqn7ahf8kii2xznqvqsk0y9qnf-apecs-physics-0.4.4.drv' failed with exit code 1; last 10 log lines:
  
  Chipmunk2D/src/cpHastySpace.c:11:10: error:
       fatal error: sys/sysctl.h: No such file or directory
         11 | #include <sys/sysctl.h>
            |          ^~~~~~~~~~~~~~
     |
  11 | #include <sys/sysctl.h>
     |          ^
  compilation terminated.
  `cc' failed in phase `C Compiler'. (Exit code: 1)
```

I ran a git bisect with range from around now to 15K commits ago, when the build still worked. First bad commit is 30286ebcc178aec1fb3797ba3ec88cf75feb282b.

<details>
<summary>Git bisect log</summary>

```
git bisect log
git bisect start
# bad: [5f04cf274984c76439ba8ed5998593b2e5a5ef7a] Merge pull request #107108 from veprbl/pr/awkward_1_0_1
git bisect bad 5f04cf274984c76439ba8ed5998593b2e5a5ef7a
# good: [8c9e7a7c7ec280f5acb496ab5ac717d82ccdf1e0] Merge pull request #86446 from gnidorah/cpu-x
git bisect good 8c9e7a7c7ec280f5acb496ab5ac717d82ccdf1e0
# good: [12d10868fcac6d3c361408ae6929447c291d6e9e] Merge pull request #97168 from r-ryantm/auto-update/hcloud
git bisect good 12d10868fcac6d3c361408ae6929447c291d6e9e
# bad: [11fc05583669cf3cb8c99dc6cb6f6ca7d93fa01d] pam_mount: add support for LUKS2
git bisect bad 11fc05583669cf3cb8c99dc6cb6f6ca7d93fa01d
# good: [8cae8e54a144897c1e58d0f48c3f9bc6740e1dc1] Revert "mediatomb/gerbera: Add release note information for 20.09"
git bisect good 8cae8e54a144897c1e58d0f48c3f9bc6740e1dc1
# bad: [582f622f6c82e8db2efc4d490e5945db177ed597] Merge pull request #101481 from makefu/pkgs/cozy/0.7.2
git bisect bad 582f622f6c82e8db2efc4d490e5945db177ed597
# bad: [1c0f102220df0b5ed01b28712579e7f76435063e] pyocr: Use tesseract without alias
git bisect bad 1c0f102220df0b5ed01b28712579e7f76435063e
# good: [6bae92a3eea8c56999d9c76417809a81e0fedfa6] Merge pull request #99980 from romildo/upd.marwaita-ubuntu
git bisect good 6bae92a3eea8c56999d9c76417809a81e0fedfa6
# good: [725a65545ece5d948c002af2c61ab35877c8f9fc] perlPackages: group aliases
git bisect good 725a65545ece5d948c002af2c61ab35877c8f9fc
# bad: [d544b64c7db02a7c8855247f8598cae10dc10e89] libheif: 1.8.0 -> 1.9.1
git bisect bad d544b64c7db02a7c8855247f8598cae10dc10e89
# bad: [25e60944c6b6c1527cebd4fe11428cf3f438b9dd] python3Packages.python-engineio: fix build w/glibc-2.32
git bisect bad 25e60944c6b6c1527cebd4fe11428cf3f438b9dd
# bad: [883bdf8070d413ae1b11d7af797db934b4eb9e29] xawtv: fix build w/glibc-2.32
git bisect bad 883bdf8070d413ae1b11d7af797db934b4eb9e29
# good: [61517de024cc9e61c18f2434656c18067a2d68a4] Merge pull request #93912 from thefloweringash/cctools-tapi
git bisect good 61517de024cc9e61c18f2434656c18067a2d68a4
# good: [97bfffe7bad60bd70e920bf738effffb7339289e] Merge pull request #97729 from marsam/update-node-14x
git bisect good 97bfffe7bad60bd70e920bf738effffb7339289e
# good: [2e075ba96c57dacec742f08636fecce11e8f5223] Merge pull request #96906 from kampka/gts
git bisect good 2e075ba96c57dacec742f08636fecce11e8f5223
# bad: [08bc71284367bf3490b0d97792dc0bc27a059ef2] dateutils: fix build w/glibc-2.32
git bisect bad 08bc71284367bf3490b0d97792dc0bc27a059ef2
# bad: [30286ebcc178aec1fb3797ba3ec88cf75feb282b] glibc: 2.31 -> 2.32
git bisect bad 30286ebcc178aec1fb3797ba3ec88cf75feb282b
# first bad commit: [30286ebcc178aec1fb3797ba3ec88cf75feb282b] glibc: 2.31 -> 2.32
```
</details>

**To Reproduce**
```
$ nix build -f '<nixpkgs>' haskellPackages.apecs-gloss
```

**Expected behavior**
Build should succeed.

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.80, NixOS, 21.03pre257780.e9158eca70a (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-21.03pre255513.8ee7c2505f9"`
 - channels(siraben): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -1541,4 +1541,8 @@ self: super: {
   # https://github.com/yesodweb/yesod/issues/1714
   yesod-core = dontCheck super.yesod-core;
 
+  # Add ApplicationServices on darwin
+  apecs-physics = addPkgconfigDepends super.apecs-physics
+    (pkgs.lib.optional pkgs.stdenv.isDarwin pkgs.darwin.apple_sdk.frameworks.ApplicationServices);
+
 } // import ./configuration-tensorflow.nix {inherit pkgs haskellLib;} self super

